xm-commons-permission/src/main/java/com/icthh/xm/commons/permission/service/PermissionContextService.java

```diff
@@ -0,0 +1,87 @@
+package com.icthh.xm.commons.permission.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.icthh.xm.commons.exceptions.BusinessException;
+import com.icthh.xm.commons.lep.LogicExtensionPoint;
+import com.icthh.xm.commons.lep.spring.LepService;
+import com.icthh.xm.commons.permission.domain.dto.PermissionContextDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static com.icthh.xm.commons.config.client.config.XmRestTemplateConfiguration.XM_CONFIG_REST_TEMPLATE;
+import static com.icthh.xm.commons.config.client.utils.RequestUtils.createAuthHeaders;
+
+@Slf4j
+@LepService(group = "service.permission")
+public class PermissionContextService {
```

Comment:
how about split to two classes
1 - generate microservice permission context
2 - handle generate accumulated permission context

single responsibility in this case will improve future maintainability